### PR TITLE
doc: Removed contractions for ESL speakers.

### PR DESCRIPTION
**Pull request by johnwilkins**
Signed-off-by: John Wilkins <jowilkin@redhat.com>

### CODE REVIEW

File: doc/cephfs/administration.rst [hunk 1]
```diff
@@ -49,7 +49,7 @@ Settings
 
     fs set <fs name> max_file_size <size in bytes>
 
-CephFS has a configurable maximum file size, and it's 1TB by default.
+CephFS has a configurable maximum file size, and it is 1TB by default.
```
Comment (tchaikov) [inline doc/cephfs/administration.rst]: i am an ESL, and i am pretty fine with "it's" or "it doesn't". just out of curiosity, is there any research result showing that contractions is lower the readability of the document from an ESL's perspective?

Comment (johnwilkins) [inline doc/cephfs/administration.rst]: This was from one of our writers in the Brno. It is to be avoided per the IBM style guide. It also makes it easier to translate.

Comment (tchaikov) [inline doc/cephfs/administration.rst]: okay, makes sense =)

**Comment by jcsp** [comment]
You mentioned the IBM style guide... did we adopt that at some stage?  If we are making changes like this, then I think we need them codified in a style guide first, and then applied to the docs second -- that way we can vet pull requests according to the style guide, and avoid regressions.

On the face of it I'm not sure we should be demanding that all doc use "I am" instead of "I'm" -- both are completely correct English, and defining a subset of the language for use in our docs is a bit of a slippery slope

**Comment by johnwilkins** [comment]
It is our style guide for downstream. So it will make upstream/downstream conversions go more smoothly. I don't intend to impose this as a requirement on anybody submitting a pull request. I wrote up a script that does the change automatically. I just did a few different pull requests to test out the script. So write with contractions if you please. That shouldn't be a problem. At Red Hat we're also exploring ways to identify weak phrases, passive voice, etc. with scripts so that it is not an imposition on the community to do anything special, and we can do editing much more efficiently.

**Comment by joscollin** [comment]
Rebase ?

**Comment by alfredodeza** [comment]
jenkins test docs

**Comment by smithfarm** [comment]
Wait, how is "it's" harder to translate than "it is"?

Compare the following sentences: (a) "Don't worry; it's OK!" (b) "Do not worry; it is OK!"

They are very different, not the same at all.

**Comment by joscollin** [comment]
I agree with @smithfarm and @jcsp suggested.
I think we don't have enough justification to accept this change. It is just to satisfy a standard set by IBM. 
Another thought I have is: Why can't the `IBM style guide` be updated ? Accept `it's` as good English.

> It also makes it easier to translate.

How ?

**Comment by wjwithagen** [comment]
@smithfarm  @joscollin @johnwilkins
I don't like the way things look once we start removing contractions, but that's just me.
And in my English classes we have been explicitly instructed to use contractions
becasue that is the way English is. Now I admit that that is a long time ago, and I haven't
payed any attention to possible language changes since.

Next "challenge" is going to be British versus American?
Is is that a non issue?

Now I can understand a little bit if this is in documentation, but what I really don't like
is that we also start language policing this in code-comments.

And I would expect only change for these kind of things when more changes are made to the same document.

**Comment by smithfarm** [comment]
@wjwithagen Yes, English is still that way - nothing has changed recently in that respect. Perhaps there are some non-native English speakers who would prefer not to have to deal with contractions, but as John said efforts to define an "acceptable" subset of English take place on a slippery slope.

With all the English weirdness and bad usage that goes on, I'm surprised that contractions - which are widely accepted as proper usage - were singled out for "correction".

All that being said, the documentation should be usable and maintainable by non-native English speakers. If the documentation maintainers really want to do this, they should be allowed to go forward.

(I'm with @wjwithagen asking that code comments be left alone, i.e. let the developers maintain them, since they are there primarily for the developers. It's already hard enough to separate substantial from insubstantial code changes when navigating the git history. Having robots tweaking English usage in the comments will make it even harder.)

**Comment by liewegas** [comment]
I'm fine with this if it eases the work of the doc writers (and other doc consumers don't object).

Needs a rebase, though.. manual-deployment was removed IIRC!

**Comment by dvanders** [comment]
FYI, Google just opened up their internal style guide, and it seems to have a more reasonable approach for contractions: https://developers.google.com/style/contractions
"It's" is permitted. "Don't" is preferred because users might miss the "not" in "do not". But noun+verb contractions are avoided, e.g. don't say "The browser's fast".

**Comment by theanalyst** [comment]
> FYI, Google just opened up their internal style guide, and it seems to have a more reasonable approach for contractions: https://developers.google.com/style/contractions
"It's" is permitted. "Don't" is preferred because users might miss the "not" in "do not". But noun+verb contractions are avoided, e.g. don't say "The browser's fast".

This sounds like a more sensible approach, also we still do not need any restriction as far as code comments go

**Comment by jcsp** [comment]
The google style guide looks very sensible.  Maybe we should just adopt that wholesale as we did for their C++ style guide!

**Comment by wjwithagen** [comment]
On 12-9-2017 16:25, Dan van der Ster wrote:
> FYI, Google just opened up their internal style guide, and it seems to
> have a more reasonable approach for contractions:
> https://developers.google.com/style/contractions
> "It's" is permitted. "Don't" is preferred because users might miss the
> "not" in "do not". But noun+verb contractions are avoided, e.g. don't
> say "The browser's fast".

Educational....
Didn't know that there were double contractions.
Hilarious to see them, wouldn't have thought of writing like that.

Others look like good suggestions. As does the rest of the guide.

--WjW

**Comment by joscollin** [comment]
Google style guide allows contractions that are commonly used. I think that's all we use commonly. I rarely find the avoided contractions in documentations or in code.

**Comment by joscollin** [comment]
No more discussions happening. Close this?

**Comment by liewegas** [comment]
@johnwilkins @smithfarm What do you think about adopting the google style guide upstream?

**Comment by smithfarm** [comment]
@liewegas I browsed it just now, and it looks well thought-out, reasoned, and based on real-life experience. For example, "Avoid trying to document future features or products, even in innocuous ways." ;-)

+1 from my side

**Comment by theanalyst** [comment]
Also CC licensed, so should be easy to adapt

**Comment by smithfarm** [comment]
Closing because https://github.com/ceph/ceph/pull/22190

**Pull request closed without merging**